Mathlib/Order/KrullDimension.lean

```diff
@@ -103,9 +121,31 @@ lemma length_le_height {p : LTSeries α} {x : α} (hlast : p.last ≤ x) :
     simp [p']
   · simp_all
 
+/--
+The height of the last element in a series is larger or equal to the length of the series.
+-/
 lemma length_le_height_last {p : LTSeries α} : p.length ≤ height p.last :=
   length_le_height le_rfl
 
+/--
+The height of an element in a series is larger or equal to its index in the series.
+-/
+lemma index_le_height (p : LTSeries α) (i : Fin (p.length + 1)) : i ≤ height (p i) :=
+  length_le_height_last (p := p.take i)
+
+/--
+In a maximally long series, i.e one as long as the height of the last element, the height of each
+element is its index in the series.
+-/
+lemma height_eq_index_of_length_eq_height_last (p : LTSeries α) (h : p.length = height p.last)
+    (i : Fin (p.length + 1)) : height (p i) = i := by
```

Comment:
Can you turn this into an iff?
```suggestion
lemma height_eq_index_of_length_eq_height_last {p : LTSeries α}
    p.length = height p.last ↔ ∀ i, height (p i) = i := by
```

Reply:
Yes, but is that useful? The other direction is a triviality and would hold for any function, not just `height`

I have a sense that you value `iff` lemmas more than I do, maybe I am missing something about their usefulness and ergonomics (besides when they are nice as rewriting rules) – what am I missing?

Reply:
They can be used in `simp` and `rw`! That's the main appeal to me

Reply:
Hmm, but rewrite rules that have equalities as their `lhs`, are they commonly useful? I wouldn’t expect, say, the equality `p.length = height p.last` to appear a lot nested inside some goal so that one would want to reach for rewriting?

I do expect that maybe one wants to rewrite `height (p i)` to `i` using `rw`, but then you does the original formulation as an implications, so that `height (p i)` is the lhs of the rule?

Reply:
> rewrite rules that have equalities as their `lhs`, are they commonly useful?

Yes, I certainly use them often

> I wouldn’t expect, say, the equality `p.length = height p.last` to appear a lot nested inside some goal so that one would want to reach for rewriting?

That might be true. I don't really know what to expect here